### PR TITLE
Feature/aimspagination

### PR DESCRIPTION
**Pull request by emelalkim**

### CODE REVIEW

File: plugins/CouchDB.js [hunk 1]
```diff
@@ -564,19 +454,87 @@ async function couchdb(fastify, options) {
...
+                    fastify
+                      .downloadAims({ aim: 'true' }, resObj, epadAuth)
+                      .then((result) => {
+                        fastify.log.info(`Zip file ready in ${result}`);
+                        const link = `http://${fastify.hostname}${result}`;
+                        // send notification and/or email with link
+                        if (request)
+                          new EpadNotification(request, 'Download ready', link, false).notify(
+                            fastify
+                          );
```
Comment (emelalkim) [inline plugins/CouchDB.js:501]: check if email is configured

File: plugins/EpadDB.js [hunk 2]
```diff
@@ -9179,8 +9366,10 @@ async function epaddb(fastify, options, done) {
               raw: true,
             });
           } else {
+            // TODO check for very large aim count (affected by pagination)
```
Comment (emelalkim) [inline plugins/EpadDB.js:9369]: count >200

File: plugins/EpadDB.js [hunk 3]
```diff
@@ -10045,10 +10236,12 @@ async function epaddb(fastify, options, done) {
             );
             fastify.log.warn('Lite project is created');
 
+            // TODO if there are more than 200 aims it is going to fail migrating!
```
Comment (emelalkim) [inline plugins/EpadDB.js:10239]: length>200

File: plugins/CouchDB.js [hunk 4]
```diff
@@ -306,169 +309,63 @@ async function couchdb(fastify, options) {
...
-            });
-            if (params.summary && params.summary.toLowerCase() === 'true') {
-              // create the csv writer and write the summary
-              const csvWriter = createCsvWriter({
-                path: `${dir}/annotations/summary.csv`,
-                header,
+            isThereDataToWrite =
+              (await fastify.prepAimDownload(
+                `${dir}/annotations`,
+                {},
```
Comment (emelalkim) [inline plugins/CouchDB.js:316]: check this is correct?

File: plugins/CouchDB.js (hunk 1 above)
Comment (emelalkim) [reply plugins/CouchDB.js:501]: done

File: server.js [hunk 5]
```diff
@@ -126,6 +126,25 @@ fastify.register(require('./routes/projectDicomweb'), { prefix: config.prefix })
 if (config.mode === 'thick') {
   fastify.register(require('./routes/plugin'), { prefix: config.prefix }); // eslint-disable-line global-require
 }
+
+fastify.register(require('fastify-nodemailer'), {
+  pool: true,
+  host: config.notificationEmail.host,
+  port: config.notificationEmail.port,
+  secure: config.notificationEmail.isTls, // use TLS
+  auth: {
```
Comment (emelalkim) [inline server.js:135]: make it an object read from config to make it more flexible

File: plugins/EpadDB.js [hunk 6]
```diff
@@ -5882,87 +5867,146 @@ async function epaddb(fastify, options, done) {
...
+              'aim_uid',
+              'dso_series_uid',
+            ],
             raw: true,
+            include: [{ model: models.project }],
           });
+          let aimUids = [];
+          let segDeletePromises = []; // an array for deleting all segs
+          if (body && Array.isArray(body)) {
+            aimUids = body;
```
Comment (emelalkim) [inline plugins/EpadDB.js:5921]: check if we are deleting segmentations with body

File: plugins/EpadDB.js [hunk 7]
```diff
@@ -5882,87 +5867,146 @@ async function epaddb(fastify, options, done) {
...
                   // eslint-disable-next-line no-await-in-loop
-                  await fastify.saveAimInternal(leftovers[i].aim_uid, request.params.project, true);
+                  await fastify.saveAimInternal(leftovers[i].aim_uid, params.project, true);
                   fastify.log.info(`Aim not deleted from system as it exists in other project`);
+                  leftoverIds.push(leftovers[i].aim_uid);
+                }
+                const deletedAims = aimUids.filter((e) => !leftoverIds.includes(e));
+                segDeletePromises = [];
+                for (let i = 0; i < deletedAims.length; i += 1) {
+                  if (deletedAims[i].dso_series_uid)
```
Comment (emelalkim) [inline plugins/EpadDB.js:5976]: deletedAims[i] doesn't have dso_series_uid

**Review by OzgeYurtsever — APPROVED** [review]
Looks good!

**Pull request merged**